ndcube/global_coords/global_coords.py

```diff
@@ -0,0 +1,72 @@
+from collections import OrderedDict
+from collections.abc import Mapping
+
+__all__ = ['GlobalCoords']
+
+
+class GlobalCoords(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+    """
+    def __init__(self, ndcube):
+        """
+        Init method.
+        """
+        super().__init__()
+        self._ndcube = ndcube
+
+        self._internal_coords = OrderedDict()
+
+    @property
+    def _all_coords(self):
+        """
+        Establish _all_coords as a property returning some _internal_coords.
+        """
+        return self._internal_coords
+
+    def add(self, name, physical_type, coords):
+        """
+        Add a new coordinate to the collection.
+        """
+        self._internal_coords[name] = (physical_type, coords)
+
+    def remove(self, name):
+        """
+        Remove a coordinate from the collection
+        """
+        del self._internal_coords[name]
+
+    @property
+    def names(self):
```

Comment:
Do we also have to implement `.keys()`?  If so, it would just be `keys = self._all_coords.keys`.  Or does the magic of `Mapping` do this for us?

Reply:
This is the same as `.keys()` to the best of my knowledge... The `names` can be updated to `keys` or be given any other name for our convenience.